solution/index.js

```diff
@@ -0,0 +1,343 @@
+'use strict'
+let taskObj, queryObj;
+const spaceAtEndRegex = /[\s]*$/g;
+let mockList = document.createElement("li");
+let rememberElement;
+
+const addEventListenerToButtons = () => {
+
+    const deleteButtons = document.querySelectorAll('.deleteButton');
+    deleteButtons.forEach(element => {
+        element.addEventListener("click", function () {
+            deleteTask(element.parentElement.parentElement.getElementsByClassName("task")[0].innerText);
+        })
+    });
+
+    const checkButtons = document.querySelectorAll('.checkButton');
+    checkButtons.forEach(element => {
+        element.addEventListener("click", function () {
+            ReplaceTaskType("done", element.parentElement.parentElement.getElementsByClassName("task")[0].innerText);
+        })
+    });
+
+    const sortButtons = document.querySelectorAll('.sortButton');
+    sortButtons.forEach(element => {
+        element.addEventListener("click", function () {
+            element.parentElement.setAttribute("draggable", "true");
+        })
+    });
+}
+
+
+const addEventListenerToTasks = () => {
+    let oldinnerTaskText;
+
+    const taskList = document.querySelectorAll('.task');
+    taskList.forEach(element => {
+        element.addEventListener("dragstart", function () { //mockList is for task drag be showen for user
+            rememberElement = element.innerHTML;
+            mockList.classList.add("mocklist");
+            mockList.innerHTML = `${element.innerText}`
+        });
+
+        element.addEventListener("dragenter", function (e) {
+            e.target.parentElement.insertBefore(mockList, e.target);
+        });
+
+        element.addEventListener("dragend", function () {
+            mockList.innerHTML = rememberElement;
+            mockList.className = "task";
+            const elementB4orAfter = mockList.previousElementSibling || mockList.nextElementSibling;
+            deleteTask(element.innerText);
+            dropTask(elementB4orAfter.innerText);
+        });
+
+        element.addEventListener("click", function () {
```

Comment:
Written function is not good and causes unwanted side effects. 
event inside event and a window level - listening